### ISSUE TITLE
Graph creation despite existing graphs

### ISSUE THREAD

**Issue by luckyhandler** [motis-project/motis#229]
We decoupled the graph creation task from the actual runnning motis instance.

The idea behind that was 
- that the service is immediately online once the container is started and 
- that the permanantly running instance doesn't need as many resources as the graph creation one.

When I run the motis instance and link it to the existing graphs I realized that it starts creating the graphs again.
Is there a parameter to tell the running instance to not do that?

**Comment by felixguendling** [comment]
MOTIS has several modules as well as the time-dependent timetable graph at its core. The time-dependent timetable graph is not on-disk but in-memory. So of course it needs to be created from the on-disk data. The loader creates a `_schedule.raw` file in the `data/schedule` directory. If you reuse this directory, the `_schedule.raw` will not be recreated.

For all other modules: they hash information about the input data (like the OSM file, `_schedule.raw`, etc.) to check if the data changed. If it changed, they repeate the preprocessing, if not then not.

If you use `dataset.begin=TODAY`, at least the `tripbased` module (if enabled) will have to do its preprocessing each day after starting the instance unless you do this in a separate preprocessing step.

So basically, I can just tell you: yes, MOTIS will reuse the files and the second start should be way faster.

This is my startup time for the Swiss dataset (first run, second run):

```bash
./motis/motis --mode test  6209,44s user 87,93s system 1314% cpu 7:59,00 total
./motis/motis --mode test  5,98s user 1,29s system 95% cpu 7,649 total
```

There is a significant speedup.

**Comment by luckyhandler** [comment]
Thanks, for me it still takes a considerable time even after setting the `dataset.cache_graph=true` flag.
Is there a healthcheck endpoint which I can check to see if the service has completely read the graphs?

**Comment by felixguendling** [comment]
Are you sure, you're reusing the data volume?

**Comment by luckyhandler** [comment]
Pretty much, yes. but not sure if the memory dump setting correctly works

**Comment by felixguendling** [comment]
I could not confirm the problem. Here's a script demonstrating the behavior:

```bash
#!/bin/bash

rm -rf data

# Create folder.
mkdir -p input/schedule

# Download and unpack MOTIS.
if [ ! -d motis ]; then
	wget https://github.com/motis-project/motis/releases/latest/download/motis-linux-amd64.tar.bz2
	tar -xf motis-linux-amd64.tar.bz2
	rm motis-linux-amd64.tar.bz2
fi

# Download and unpack timetable.
rm -rf input/schedule
wget https://opentransportdata.swiss/de/dataset/timetable-2022-gtfs2020/permalink -O swiss.zip
unzip swiss.zip -d input/schedule
rm swiss.zip

# Download OSM input
if [ ! -f input/osm.pbf ]; then
	wget https://download.geofabrik.de/europe/switzerland-latest.osm.pbf -O input/osm.pbf
fi

# Write config.ini
cat << EOF > input/config.ini
modules=routing

server.static_path=motis/web
dataset.cache_graph=true

[import]
paths=schedule:input/schedule
paths=osm:input/osm.pbf
data_dir=data

[osrm]
profiles=motis/osrm-profiles/car.lua
profiles=motis/osrm-profiles/bike.lua
profiles=motis/osrm-profiles/bus.lua

[ppr]
profile=motis/ppr-profiles/default.json

[tiles]
profile=motis/tiles-profiles/background.lua
EOF

# Start MOTIS
time ./motis/motis -c input/config.ini --mode test > /dev/null
time ./motis/motis -c input/config.ini --mode test > /dev/null
```

Output:
```
[info][2022-03-29T09:26:20Z+0000][main.cc:165] system boot finished
[info][2022-03-29T09:26:21Z+0000][main.cc:169] shutdown

real	0m25,628s
user	0m20,427s
sys	0m4,491s
[info][2022-03-29T09:26:22Z+0000][main.cc:165] system boot finished
[info][2022-03-29T09:26:23Z+0000][main.cc:169] shutdown

real	0m1,969s
user	0m0,834s
sys	0m0,136s
```

The boot time goes down from ~25s to ~1s.

Please post a script that illustrates the problem. Without a minimal example script it's not possible to reproduce your problem.

**Comment by luckyhandler** [comment]
What takes so long is the `tripbased` and the `path` reading

**Comment by felixguendling** [comment]
Please try this script and report your results:

```
#!/bin/bash

rm -rf data

# Create folder.
mkdir -p input/schedule

# Download and unpack MOTIS.
if [ ! -d motis ]; then
	wget https://github.com/motis-project/motis/releases/latest/download/motis-linux-amd64.tar.bz2
	tar -xf motis-linux-amd64.tar.bz2
	rm motis-linux-amd64.tar.bz2
fi

# Download and unpack timetable.
rm -rf input/schedule
wget https://opentransportdata.swiss/de/dataset/timetable-2022-gtfs2020/permalink -O swiss.zip
unzip swiss.zip -d input/schedule
rm swiss.zip

# Download OSM input
if [ ! -f input/osm.pbf ]; then
	wget https://download.geofabrik.de/europe/switzerland-latest.osm.pbf -O input/osm.pbf
fi

# Write config.ini
cat << EOF > input/config.ini
modules=routing
modules=path
modules=tripbased
modules=osrm

server.static_path=motis/web
dataset.cache_graph=true

[import]
paths=schedule:input/schedule
paths=osm:input/osm.pbf
data_dir=data

[osrm]
profiles=motis/osrm-profiles/bus.lua

[ppr]
profile=motis/ppr-profiles/default.json

[tiles]
profile=motis/tiles-profiles/background.lua
EOF

# Start MOTIS
time ./motis/motis -c input/config.ini --mode test > /dev/null
time ./motis/motis -c input/config.ini --mode test > /dev/null
```

**With `tripbased` and `path enabled`:**

First run:
```
real	1m57,701s
user	19m4,767s
sys	0m30,580s
```

Second run:
```
real	0m2,450s
user	0m1,070s
sys	0m0,381s
```

You don't seem to reuse your data directory or you repeat this at another day with `dataset.begin=TODAY` so that the graph is not valid anymore.
